pkg/repository/backend_postgres.go

```diff
@@ -823,6 +823,37 @@ func (c *PostgresBackendRepository) GetDeploymentByNameAndVersion(ctx context.Co
 	return &deploymentWithRelated, nil
 }
 
+func (c *PostgresBackendRepository) ListLatestDeploymentsWithRelatedPaginated(ctx context.Context, filters types.DeploymentFilter) (common.CursorPaginationInfo[types.DeploymentWithRelated], error) {
+	query := squirrel.StatementBuilder.PlaceholderFormat(squirrel.Dollar).
```

Comment:
Looks like we're using a question mark instead of the dollar. Can we update this to Question and also confirm that we're only getting data for the provided workerspace id?